### DISCUSSION TITLE
What are the guiding questions of this project?

### DISCUSSION THREAD

**Discussion by galengorski** [USGS-R/drb-do-ml#8] [Ideas]
In discussing some of the initial decisions about the structure and direction of this project, I thought it might be helpful to come up with a list of guiding/motivating questions. These could be changed and altered as we move forward with the project and talk to collaborators, but it might help to keep us focused on the goals of the project. Below are some potential questions, which we can add to or whittle down if need be:

How can we use PGDL of DO:
- to improve the accuracy of predictions of in-stream DO in the DRB?
- to develop models that generalize better than process-based models outside of the region (or general parameter space) in which they were trained?
- with a transfer learning approach to make predictions in areas (or during time periods) with sparse data?
- to better understand the underlying physical processes that drive in-stream DO?
- to address existing gaps in understanding of fundamental processes governing DO dynamics?

**Comment by jsadler2** [comment]
Another one that @lekoenig brought up in our meeting last week was "What's the value of spatially-distributed, temporally-sparse samples compared to spatially-sparse, temporally-continuous samples?" This was prompted by her DO data inventory
![image](https://user-images.githubusercontent.com/6943441/140782564-dc97d428-55e6-41ce-9833-72f6a0879ef6.png)

**Reply by lekoenig** [reply]
To follow up on Jeff's comment, I was mostly thinking about predictive accuracy here, but I *think* it's a separate question to also ask about the relative value of these datasets for inferring causality, perhaps with an eye towards @galengorski's information theory work.

**Reply by galengorski** [reply]
I have been drafting up an example workflow for how I think information theory could be used to assess model functional performance, mostly as it relates to the outputs of process based models for the estuary salinity project. I'd be happy to show it to you at a meeting. It might help us think about applications in the future.

**Comment by lekoenig** [comment]
Adding to our guiding questions here a thought that @jsadler2 and I discussed a bit yesterday: DO itself is really only interesting when it gets too low, so the value of making accurate predictions is arguably not the same across the linear range of DO concentrations we'd expect in rivers. 

We didn't make any modeling decisions, but we might keep this in mind as we develop our plans (and get input from collaborators and the ML-DA task group, too) - for example, maybe we should incorporate some low-oxygen concentration (2-3 mg/L) as a benchmark statistic in the model, or think about whether/how we would carry out the train/test splits to optimize predictive accuracy within a threshold DO range.

**Reply by galengorski** [reply]
That's an interesting conversation, thanks for sharing. Alison brought [this paper](https://hess.copernicus.org/preprints/hess-2021-423/) up at a previous meeting, but I put it in our shared folder under ML-Applications. It looks at trying to tailor the train/test split to better predict extreme events. It doesn't work, but it has some interesting ideas in it I think.

On a related note to the guiding questions. Is our broad goal to have a single model that predicts DO well everywhere? I just wonder if low DO is a priority, and there is a potential tradeoff with getting good predictions across a range of values maybe developing multiple models might allow us to mitigate the tradeoff a bit.
